cmd/vela-worker/exec.go

```diff
@@ -37,9 +38,16 @@ func (w *Worker) exec(index int) error {
 	}
 
 	// retrieve a build token from the server to setup the execBuildClient
-	bt, _, err := w.VelaClient.Build.GetBuildToken(item.Repo.GetOrg(), item.Repo.GetName(), item.Build.GetNumber())
+	bt, resp, err := w.VelaClient.Build.GetBuildToken(item.Repo.GetOrg(), item.Repo.GetName(), item.Build.GetNumber())
 	if err != nil {
 		logrus.Errorf("unable to retrieve build token: %s", err)
+
+		// build is not in pending state — user canceled build while it was in queue. Pop, discard, move on.
+		if resp != nil && resp.StatusCode == http.StatusBadRequest {
```

Comment:
Is there any other reason we might get a 400 error here?

Reply:
Yeah. If this returned a code that is unique to "the build is not runnable" (not in pending or running states) that would be great.

Maybe one of
- 409 Conflict https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409
- 410 Gone https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/410